### DISCUSSION TITLE
how to close go-agent log & metrics report

### DISCUSSION THREAD

**Discussion by blanche789** [apache/skywalking#12271] [Q&A] [answered]
![image](https://github.com/apache/skywalking/assets/38674578/85df4565-25b5-4a85-8b85-f0709dc85bab)

When using go-agent to monitor a Go application, the error shown in the log mentioned above occurred. I only want to report traces, but I don't want to report logs and metrics. Is there no way to disable them?

**Comment by mrproliu** [comment]
For now, you can disable to report the log reporter [following this documentation](https://skywalking.apache.org/docs/skywalking-go/next/en/advanced-features/logging-setup/#log-upload), but metrics we don't have a configuration for disabling it.

**Reply by blanche789** [reply]
Is there a plan to provide support for this in the future? If it can't be disabled, will it affect the performance of the agent?

**Reply by mrproliu** [reply]
> Is there a plan to provide support for this in the future? 

For now, we don't have a plain to support it. 

>  If it can't be disabled, will it affect the performance of the agent?

Nope, I will not affect the performance.

**Reply by mrproliu** [reply, marked as answer]
The metrics reporter [reports them by interval](https://github.com/apache/skywalking-go/blob/7948b433093a6ff51a7131a8f86ecfe5fd9ed56c/tools/go-agent/config/agent.default.yaml#L27-L29).

**Reply by blanche789** [reply]
Can I add a switch to the metrics myself?

**Reply by wu-sheng** [reply]
CAN? This is an open-source project, you have the source codes, you could do anything you like.

**Reply by blanche789** [reply]
ok, thanks
